### ISSUE TITLE
jellyfin on roku - no "current channel" indicator on channel guide on startup

### ISSUE THREAD

**Issue by cpankonien** [jellyfin-archive/jellyfin-roku-legacy#336]
On Jellyfin Live TV start on Roku, there is no indicator to show what channel you are actually on.

 - OS: (server) Ubuntu 20.04.1 LTS
 - Virtualization: none
 - Clients: Roku 4200x, 4620x
 - Browser: n/a
 - Jellyfin Version: 10.6.4
 - Playback: n/a 
 - Installed Plugins: AudioDB, MusicBrainz, OMDb, TheTVDB
 - Reverse Proxy: none
 - Base URL: n/a
 - Networking: roku wifi, server ethernet
 - Storage: server storage local

1. Open Jellyfin on Roku device
2. Open "Live TV"
3. Observe that there is no current channel indicator

On opening Live TV, one would expect to see what channel you are currently on and then be able to scroll up/down/right, etc.

To reproduce:  Open Jellyfin on Roku device, then hit Live TV. Hit the asterisk button on Roku remote (options), click "view" and then scroll down to "TV Guide" and hit "OK" (even if it's already checked). Hit the back arrow (upper left-most button on remote) twice, blue box should now be there, indicating the currently selected channel.

**Comment by neilsb** [comment]
Could you let me know what you do see please?  Do you see the channel information, and if you scroll around do the program details at the top of the screen change, and the channel list and time along the top change too?  

I am unable to reproduce this on the 2 devices and 3 servers I have access to, and the Blue Highlight bar always appears when the guide is finished loading:

![image](https://user-images.githubusercontent.com/5013482/101921773-45c3fb80-3bc5-11eb-8a45-fec71e5b9100.png)

**Comment by cpankonien** [comment]
![DSC_3013](https://user-images.githubusercontent.com/62761551/101952581-e109a080-3bc6-11eb-8495-c4691153af65.JPG)
I see the channels listed on the left side as well as specific program data under each time slot. If I arrow up, down and right, I hear a (normal) click sound but nothing on the screen changes. If I do the aforementioned procedure, it works as expected. The blue highlight bar does not appear until I do the series of button presses mentioned earlier.

**Comment by cpankonien** [comment]
I tried this on another Roku (3920x) and it appears to work normally...box is highlighted on start, etc. I assumed (yea, I know) that Rokus were pretty much the same, at least as far as code goes, but maybe not!

**Comment by neilsb** [comment]
> The blue highlight bar does not appear until I do the series of button presses mentioned earlier.

Sorry - possibly me being dumb, but could you clarify how you get the blue bar to appear again?

Can you also let me know the default landing page you have set for LiveTV in the web options  (i.e. When selecting Live TV, do you goto the Channel List or TV Guide by default)?

**Comment by cpankonien** [comment]
Open Jellyfin on Roku device, then hit Live TV. Hit the asterisk button on Roku remote (options), click "view" and then scroll down to "TV Guide" and hit "OK" (even if it's already checked). Hit the back arrow (upper left-most button on remote) twice, blue box should now be there, indicating the currently selected channel.

When I select "Live TV" I land on the TV Guide. I did not see an option to set Channel List, only "Guide" or "Suggestions". (That is in User>Settings>Home>Live TV>Default Screen>Guide or Suggestions)

I wasn't aware that I culd actually change all of those things, but I won't mess with them now to avoid any confusion!

**Comment by digi-ghost** [comment]
I'm also seeing the same behaviour. Go into Live TV and there's no selection indicator, but using the arrow keys I hear the clicks, the screen doesn't change, but pressing OK will take me to a channel. I also discovered the same method as @cpankonien to get it to show up - asterisk, view, choose Guide (even if that's already selected), back button.

In my options from the web site I have guide set as the default view for Live TV.

- OS: (server) Windows 10
- Virtualization: none
- Clients: Hisense Roku TV
- Browser: n/a
- Jellyfin Version: 10.6.4
- Playback: n/a
- Installed Plugins: AudioDB, MusicBrainz, OMDb, TheTVDB
- Reverse Proxy: none
- Base URL: n/a
- Networking: Roku TV ethernet, server ethernet
- Storage: server storage local, NAS

**Comment by adgoncal** [comment]
I am running into the same issue with my Hisense Roku TV (55R6040G).
My server is running the `jellyfin/jellyfin` (10.6.4) docker container and live TV is using the TVHeadend Plugin (6.0.0.0)

**Comment by neilsb** [comment]
I think this happened to me once, while I was looking into something else.   I cannot reproduce this consistently, so fixing the issue is not straightforward.

To those of you having this issue, is it happening consistently?   Does it ever **not** have this issue?

My best guess is that the focus is not being set correctly on the Grid.  However, this is set when this page shows, so should always bet the case when schedule data is loaded.  I'd like to force the Grid to be the focus again when the schedule data is loaded but without being able to reproduce this consistently I would know if it has any effect.

If any of you do consistently get it, we could try this fix on the beta channel if you'd be happy to install that, to allow us to get feedback.

**Comment by cpankonien** [comment]
Yes, it is quite consistent. Happens every time, without fail. I'd be glad to use the beta version, will install it now.

Beta installed, but the only option available is the channel listing, can't seem to get to the channel guide.

After selecting "Live TV", the only option available after hitting * (options) is "Sort by>Name" and "Sort Order>Ascending", neither of which have any other modes for selection.

**Comment by neilsb** [comment]
Bit of a delay, but a beta release has been made which has an update to try and resolve this. Still waiting for the latest version to be available from the website (currently still showing 1.3.5)

**Comment by neilsb** [comment]
Some initial problems with the update to the Beta version being available for download, but that should be it now (v1.4.7)

**Comment by neilsb** [comment]
@cpankonien Latest store version now has a change to try and force the channel indicator.  Since I cannot replicate this, would you be able to test the latest store version (1.4.7) and let me know if the issue still exists?

Thanks

**Comment by cpankonien** [comment]
Sorry, no longer using jellyfin, had to switch back to plex. Just too many
issues.

**Comment by neilsb** [comment]
@digi-ghost @adgoncal would either of you be able to see if the latest store (1.4.7) release has resolved this issue for you please?

@cpankonien No problem. Thanks for letting me know. Hope to see you back someday 😊

**Comment by adgoncal** [comment]
When I first posted in this thread, when I clicked Live TV I was taken to the guide. Now, every time I click Live TV I am taken to the channels list. If I press `*` > `View` > `TV Guide`, it takes me to the guide and the channel is highlighted as expected.

Take my experience with a grain of salt:
1. I am using TVHeadend plugin
2. Right now Live TV is not playing on my RokuTV when using Jellyfin. It gets stuck "retrieveing". I have not debugged this yet, but I think it has to do with my transcoding settings, and the codecs available on this Hisense Roku TV (55R6040G).

**Comment by anthonylavado** [comment]
@adgoncal To change the default appearance when entering Live TV, you need to change your profile's setting on the web interface.

When logged in to Jellyfin Web:
- Click on your profile icon at the top right
- Choose "Home"
- Scroll down to Live TV, and change the default screen to "Guide"
- Scroll down all the way and click Save

**Comment by adgoncal** [comment]
@anthonylavado I can confirm after changing the default to "Guide", every time I open Live TV the program is highlighted as expected. Thank you!

**Comment by neilsb** [comment]
Will just close this for now since indications are that it's fixed. Just reopen the issue if there's still a problem.

**Comment by steamyRayV** [comment]
Hello! I decided to check the progress of Jellyfin on Roku and the guide is unusable with this bug. I can't find any version info within the roku app but the Roku claims it's up to date. Server version is 10.8.4. 
Using the "On Now" menu I am able to select a channel to play that work in Emby, but it does not stream in Jellyfin so that's another issue.
At least with my equipment, the guide on Roku and live streaming on Roku are not functional at all.

EDIT: I got it to play a recording from Emby and the client version is 1.5.0, according to the server dashboard.
